### DISCUSSION TITLE
Correct Errors

### DISCUSSION THREAD

**Discussion by jrbarnes9** [ONSdigital/design-system#30] [Feedback]
Use this place to discuss the [correct errors pattern](https://ons-design-system.netlify.com/patterns/error-validation/) in the [ONS Design System](https://ons-design-system.netlify.com)

**Comment by jrbarnes9** [comment]
Proposing to highlight empty fields with a red outline:

<img width="644" alt="Screenshot 2020-05-06 at 08 10 09" src="https://user-images.githubusercontent.com/43346934/81146950-34e94280-8f71-11ea-92f2-a80f581a0e2f.png">

**Reply by jrbarnes9** [reply]
#782 resolved

**Comment by jrbarnes9** [comment]
List numerals should use a monospaced variant of the font to ensure lists line up...
![Screenshot 2020-06-18 at 08 46 55](https://user-images.githubusercontent.com/43346934/84993542-537e5400-b141-11ea-8820-1cd2db183430.png)

**Reply by jrbarnes9** [reply]
#783 resolved

**Comment by jrbarnes9** [comment]
Highlight write-in when input fails validation
![Screenshot 2020-07-10 at 10 18 20](https://user-images.githubusercontent.com/43346934/87138701-b8d3e980-c296-11ea-9506-7edeb5260c1b.png)

![Screenshot 2020-07-10 at 10 19 09](https://user-images.githubusercontent.com/43346934/87138758-d0ab6d80-c296-11ea-9606-f58f5b28728b.png)

**Reply by jrbarnes9** [reply]
#826 resolved

**Comment by jrbarnes9** [comment]
Error summary list numerals are not vertically aligned with links

1px out.
<img width="618" alt="Screenshot 2020-07-24 at 15 42 37" src="https://user-images.githubusercontent.com/43346934/88403595-ac867b00-cdc4-11ea-9faa-13bab7fd68e6.png">

**Reply by jrbarnes9** [reply]
#853 resolved

**Comment by jrbarnes9** [comment]
Error summary list numerals should not be hidden from screen readers. 

Currently have attribute `aria-hidden=true`

**Reply by jrbarnes9** [reply]
#854 resolved

**Comment by jrbarnes9** [comment]
Error summary should use ordered lists

**Reply by jrbarnes9** [reply]
#855 resolved

**Comment by jrbarnes9** [comment]
Error summary panel heading isn't marked up as a heading.  (probably should be an `h1`) See: https://github.com/alphagov/govuk_elements/pull/510

**Reply by jrbarnes9** [reply]
#1079 resolved

**Comment by jrbarnes9** [comment]
Document validation error page changes:
- Prefixing `<title>` with 'Error:'
- Keyboard focus moved to error summary panel using `scrollIntoView()`

**Reply by jrbarnes9** [reply]
Issue #1330 raised

**Comment by jrbarnes9** [comment]
Don't highlight validation error on write-in input unless input is mandatory and radio/checkbox is selected

The below example write-in input field shouldn't have a red border because it's radio hasn't been selected.  Only if the radio is selected, and the write-in input is mandatory, should the input border be highlighted.  

Screenshot of [pattern example prototype](https://ons-design-system.netlify.app/patterns/error-validation/examples/errors-proto/errors/?other-answer=&day=&month=&year=&btn-submit=)
![Screenshot 2021-03-31 at 08 47 11](https://user-images.githubusercontent.com/43346934/113109611-0c5bde00-91fe-11eb-9726-5d996fad1162.png)

**Reply by jrbarnes9** [reply]
#1461 resolved

**Comment by jrbarnes9** [comment]
Don't use a list for error summary if only one error on page

Show just a link...
![Screenshot 2021-03-31 at 09 21 20](https://user-images.githubusercontent.com/43346934/113114036-db31dc80-9202-11eb-89b6-318c61ac51c4.png)

**Reply by jrbarnes9** [reply]
#1462 resolved

**Comment by jrbarnes9** [comment]
Incorrect input label in correct errors pattern example prototype
![Screenshot 2021-04-08 at 12 22 14](https://user-images.githubusercontent.com/43346934/114018434-22544900-9865-11eb-9d57-bd74f954af5a.png)

**Reply by jrbarnes9** [reply]
#1472 resolved

**Comment by Paul-Joel** [comment]
I'm getting the odd wrapping behaviour again in the error panel
![image](https://user-images.githubusercontent.com/19411155/129187096-b2a2c680-88fe-41a8-84e7-4c94768f9494.png)

Generated panel code is
```
<div class="panel__body">
    <p class="">
                <span class="list__prefix" aria-hidden="true">1.
                    </span><a href="#error_mobile" class="list__link js-inpagelink" data-qa="list-item-1">Enter a UK mobile number in a valid format, for example, 07700 900345 or +44 7700 900345</a></p>

            </div>
```
Currently using 36.0.0

**Reply by jrbarnes9** [reply]
Resolved by removing `'index': true` from macro

**Comment by jrbarnes9** [comment]
Input not focussed after selecting error summary panel link

**Reply by jrbarnes9** [reply]
#1739 raised

**Comment by jrbarnes9** [comment]
The mutually exclusive components in the correct errors prototype are broken

**Reply by jrbarnes9** [reply]
Raised #2279
